### DISCUSSION TITLE
Error when using repository variable for timeout-minutes in Github Actions

### DISCUSSION THREAD

**Discussion by sgrtye** [community/community#55332] [Actions] [answered]
### Select Topic Area

Bug

### Body

I am trying to have some default variables across multiple workflows using repository variables.

`jobs:
  Template:
    runs-on: ${{ vars.SYSTEM }}
    timeout-minutes: ${{ vars.TIMEOUT }}`

And an error occurs with the message 
[Template: .github#L1](https://github.com/sgrtye/Actions/commit/30a2886e69aeb64f711d4900e7c2479b3d85bcc7#annotation_11220353181)
Error when evaluating 'timeout-minutes' for job 'Template'. .github/workflows/action-Template.yml (Line: 8, Col: 22): Unexpected value '10'

The variable vars.SYSTEM was interpreted just fine but not the vars.TIMEOUT.

Would anyone be able to help with this?

**Comment by davidoliver-1** [comment]
The error you are encountering when using a repository variable for timeout-minutes in Github Actions is likely due to the fact that timeout-minutes requires a numeric value. If the repository variable you are using is not numeric or is not being passed as a numeric value, you will receive an error.
same error i got in [apkbeyond](https://apkbeyond.com/)

**Reply by airtower-luna** [reply, marked as answer]
If they really missed type conversion there, it should be possible to work around the problem with [`fromJSON()`](https://docs.github.com/en/actions/learn-github-actions/expressions#fromjson), because a string containing only a number is valid JSON to express that number. Try: `timeout-minutes: ${{ fromJSON(vars.TIMEOUT) }}`

That would be a rather absurd bug. :smile_cat:

**Reply by sgrtye** [reply]
Thank you both for your timely reply. The code suggested does solve the conversion problem. 

Neither the actions.yml file nor the repository variable has an explicit typing system, let alone type conversion. :clown_face:

**Reply by airtower-luna** [reply]
YAML does have a type system. For example, `a` is a number and `b` a string:

```yaml
a: 5
b: '5'
```

What I suspect is that they only tested with numeric literals and didn't consider that anyone using Actions substitutions would end up creating a string (because to the YAML parser `${{ vars.TIMEOUT }}` is a string, not a number, and so are the `vars` items) that might need to be converted to a number after variable substitution. Fun!

Might be [worth a bug report](https://github.com/actions/runner/issues), in case the staff members around here miss this (to me) highly entertaining thread. :sweat_smile: 

There's plenty of weirdness to YAML without mixing in Actions, though. I'll just leave this link here: [The yaml document from hell](https://ruudvanasseldonk.com/2023/01/11/the-yaml-document-from-hell) :smile_cat:

**Reply by jnguyen75** [reply]
Ran into a similar issue and had to workaround this by coercing the type to a string, only to be later converted back to a number in our scripts.

Was a bug ever reported for this issue?

**Comment by arne-romo** [comment]
See https://docs.github.com/en/actions/learn-github-actions/expressions#example-returning-a-json-data-type
```
name: print
on: push
env:
  continue: true
  time: 3
jobs:
  job1:
    runs-on: ubuntu-latest
    steps:
      - continue-on-error: ${{ fromJSON(env.continue) }}
        timeout-minutes: ${{ fromJSON(env.time) }}```
